types/rlp_unmarshal_storage.go

```diff
@@ -67,46 +104,113 @@ func (t *Transaction) UnmarshalStoreRLPFrom(p *fastrlp.Parser, v *fastrlp.Value)
 		return err
 	}
 
-	if len(elems) != 2 {
-		return fmt.Errorf("expected 2 elements")
+	if len(elems) != 2 && len(elems) != 3 {
+		return fmt.Errorf("expected 2 or 3 elements")
 	}
 
-	// consensus part
-	if err := t.UnmarshalRLPFrom(p, elems[0]); err != nil {
-		return err
-	}
-	// context part
-	if err = elems[1].GetAddr(t.From[:]); err != nil {
-		return err
+	if len(elems) == 2 {
+		// consensus part
+		if err := t.UnmarshalRLPFrom(p, elems[0]); err != nil {
+			return err
+		}
+		// context part
+		if err = elems[1].GetAddr(t.From[:]); err != nil {
+			return err
+		}
+	} else if len(elems) == 3 {
+		// consensus part
+		if elems[0].Type() != fastrlp.TypeBytes {
+			return fmt.Errorf("expected ByteType")
+		}
+
+		bytes, err := elems[0].Bytes()
+		if err != nil {
+			return err
+		}
+
+		if l := len(bytes); l != 1 {
+			return fmt.Errorf("expected 1 byte transaction type, but size is %d", l)
+		}
+
+		if t.Type, err = ToTransactionType(bytes[0]); err != nil {
+			return err
+		}
+
+		if err := t.UnmarshalRLPFrom(p, elems[1]); err != nil {
+			return err
+		}
+
+		// context part
+		if err = elems[2].GetAddr(t.From[:]); err != nil {
+			return err
+		}
 	}
 
+	t.ComputeHash()
+
 	return nil
 }
 
 func (r *Receipts) UnmarshalStoreRLP(input []byte) error {
 	return UnmarshalRlp(r.UnmarshalStoreRLPFrom, input)
 }
 
+//nolint:dupl
```

Comment:
Can we extract the common areas of these methods that are duplicated, and drop this `nolint`?

Reply:
Created method in TransactionType 775e5f00b9656f90fdeded9eb408867d95b37a38